### PR TITLE
[3.7] fix moveshard, when missing current entry to a shard

### PR DESCRIPTION
**Pull request by kvahed**
### Scope & Purpose

Moveshard has a couple of iterators operating on vpack arrays, which were hardened against non array slices.
This is a fix detected in chaos tests in devel and 3.7 branches

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [ ] :book: CHANGELOG entry made

#### Backports:

- [ ] No backports required
- [ ] Backports required for: *(Please specify versions and link PRs)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket number:
- [ ] Design document: 

### Testing & Verification

*(Please pick either of the following options)*

- [ ] This change is a trivial rework / code cleanup without any test coverage.
- [x] The behavior in this PR was *manually tested*
- [ ] This change is already covered by existing tests, such as *(please describe tests)*.
- [X] This PR adds tests that were used to verify all changes:
  - [X] Added new C++ **Unit tests**
  - [ ] Added new **integration tests** (e.g. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/14298/

### Documentation

> All new features should be accompanied by corresponding documentation. 
> Bugs and features should furthermore be documented in the CHANGELOG so that
> support, end users, and other developers have a concise overview. 

- [ ] Added entry to *Release Notes* 
- [ ] Added a new section in the *Manual* 
- [ ] Added a new section in the *HTTP API* 
- [ ] Added *Swagger examples* for the HTTP API  
- [ ] Updated license information in *LICENSES-OTHER-COMPONENTS.md* for 3rd party libraries

### External contributors / CLA Note 

Please note that for legal reasons we require you to sign the [Contributor Agreement](https://www.arangodb.com/documents/cla.pdf)
before we can accept your pull requests.

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -1,6 +1,8 @@
 v3.7.10 (XXXX-XX-XX)
 --------------------
 
+* Resonably harden MoveShard against VelopcyPack more.
```
Comment (jsteemann) [inline CHANGELOG:4]:
```suggestion
* Resonably harden MoveShard against unexpected VelocyPack input.
```

File: arangod/Agency/MoveShard.cpp [hunk 2]
```diff
@@ -279,28 +279,32 @@ bool MoveShard::start(bool&) {
...
+      }
+      ++count;
     }
-    if (srv.copyString() == _from) {
-      found = count;
-    }
-    ++count;
+  } else {
+    // We really should never be ever here
+    moveShardFinish(false,  false, std::string("shard ") + _shard + "has no planned database servers");
```
Comment (jsteemann) [inline arangod/Agency/MoveShard.cpp:306]:
```suggestion
    moveShardFinish(false,  false, std::string("shard ") + _shard + " has no planned database servers");
```

File: arangod/Agency/MoveShard.cpp [hunk 3]
```diff
@@ -279,28 +279,32 @@ bool MoveShard::start(bool&) {
...
+        found = count;
+      }
+      ++count;
     }
-    if (srv.copyString() == _from) {
-      found = count;
-    }
-    ++count;
+  } else {
+    // We really should never be ever here
```
Comment (jsteemann) [inline arangod/Agency/MoveShard.cpp:305]: I think we should adjust this code comment, as it is misleading. If we should *never* be here, why is this code triggered in some edge case? We should probably rephrase this to sth like "under normal circumstances we should not get here, but in the very unlikely case of ____ this code gets triggered"

File: arangod/Agency/MoveShard.cpp [hunk 4]
```diff
@@ -814,19 +890,28 @@ JOB_STATUS MoveShard::pendingFollower() {
...
-                             trx.add(srv);
+                         if (plan.isArray()) {
+                           for (VPackSlice srv : VPackArrayIterator(plan)) {
+                             if (srv.copyString() != _from) {
+                               trx.add(srv);
+                             }
                            }
+                         } else {
+                           LOG_TOPIC("dbc18", WARN, Logger::SUPERVISION)
+                             << "failed to iterator of planned servers for shard "
```
Comment (jsteemann) [inline arangod/Agency/MoveShard.cpp:910]:
```suggestion
                             << "failed to iterate over planned servers for shard "
```

**Review by jsteemann — APPROVED** [review]

File: arangod/Agency/MoveShard.cpp (hunk 3 above)
Comment (kvahed) [reply arangod/Agency/MoveShard.cpp]: done
